lib/chef/knife/bmcs_server_list.rb

```diff
@@ -25,10 +26,42 @@ def run
         options = {}
         options[:limit] = config[:limit] if config[:limit]
 
+        servers = {}
+
         response = compute_client.list_instances(compartment_id, options)
 
-        display_list(response, ['Display Name', 'ID', 'State']) do |item|
-          [item.display_name, item.id, item.lifecycle_state]
+        if config[:format] == 'summary' && response.data
+          # build a hash of hashs, keyed off of server_id
+          response.data.each do |server|
+            vnics = compute_client.list_vnic_attachments(compartment_id, instance_id: server.id)
+            vnics.data.each do |vnic|
+              servers[vnic.instance_id] = {}
+              servers[vnic.instance_id]['vnic_id'] = vnic.vnic_id
+              if vnic.lifecycle_state == 'ATTACHED'
+                begin
+                  vnic_info = network_client.get_vnic(vnic.vnic_id, {})
+                rescue OracleBMC::Errors::ServiceError
+                  servers[vnic.instance_id]['private_ip'] = '(none)'
```

Comment:
In this case it's not necessarily none, but may be that you don't have permission. Rather than showing separate messages for each of none, no permissions, and detached, I think it would be reasonable to just leave the field blank in every case where we don't get an address.